### ISSUE TITLE
Custom API Routes

### ISSUE THREAD

**Issue by kcastro18** [OrchardCMS/Orchard#7973]
Hi!

I have a custom module with custom api routes that I created. In my Routes.cs file I declared a route to http://domain/lawyers and it's working locally in visual studio/iis express just fine. When I tried to deploy it IIS server the I can't get to the lawyers page. It saying "Not found. The page you are looking for does not exist."

I've tried everything I can find to find where the problem is and tried the solutions suggested in stackoverflow but I still can't make it work in the IIS server. Please help.

Thanks,
Karen

**Comment by sfmskywalker** [comment]
Anything suspicious in your log files (App_Data/Logs)? I have seen 404's due to tenants not starting up due to incorrect connection string values. If that's the case, you'll see DB related errors in the log file.

**Comment by kcastro18** [comment]
There are no errors in the log files. That is why I'm having a hard time figuring it out because I don't know where to look. I'm able to connect to the db and it shows the pages I created before. But it seems like my custom modules are not working. I wrote a lawyer module with a lawyer content type that takes a Bar ID and shows the lawyers information. But when I run orchard in the server the Bar ID input is not showing and it works fine locally running it via Visual Studio/IISExpress. Not sure if that is related to my first problem but I hope it helps.

**Comment by Hazzamanic** [comment]
Just checking... have you enabled your custom modules on your server?

**Comment by kcastro18** [comment]
Yes I did. And it ran the data migrations too because I saw the tables in db.

**Comment by kcastro18** [comment]
I tried doing a clean install of orchard in a new empty database then I re-publish my Orchard.Web project that includes my custom module and copying it to the new website without the App_Data folder. I enabled the modules in the admin UI and now nothing is happening. I don't see the tables that the modules is supposed to create.

**Comment by kcastro18** [comment]
Does anyone have any ideas how to fix this? I am using Visual Studio 2017. The first time I deployed my custom modules in the server it works fine and I was using VS 2015 at that time. But I had to upgrade my pc to Windows 10 and installed a VS 2017. Since I used VS 2017 this problem occurred. So I am not sure if it's VS 2017 that has a problem or what. Please help. Thanks in advance.

**Comment by kcastro18** [comment]
I found the problem. So when building and publishing the Orchard.Web, it didn't include the assemblies of my custom modules and I don't know why. I created the modules using Orchard.exe command prompt and it should by default include the custom modules' precompiled  assemblies in the deployment, if I understood it correctly. The workaround that I did is referenced my custom modules in Orchard.Web project then redeploy it and it worked. So glad I noticed that the assemblies are missing.

I wonder if that is a bug though? It would be great if anyone can shed some light with this issue.
